src/metpy/plots/declarative.py

```diff
@@ -656,6 +660,39 @@ class MapPanel(Panel):
     'xx-small', 'x-small', 'small', 'medium', 'large', 'x-large', 'xx-large'.
     """
 
+    @validate('area')
+    def _valid_area(self, proposal):
+        """Check that proposed string or tuple is valid and turn string into a tuple extent."""
+        area = proposal['value']
+
+        # Parse string, check that string is valid, and determine extent based on string
+        if isinstance(area, str):
+            try:
+                region, modifier = re.match(r'(\w+)([-+]*)$', area).groups()
+            except AttributeError:
+                raise TraitError('"' + area + '" is not a valid string area.')
```

Comment:
Technically this try...except would catch any uncaught `AttributeError` happening anywhere while executing that line. While this definitely works, and a weird `AttributeError` bubbling up is unlikely, I feel like explicitly checking for `None` (which is what is documented as the behavior for "no match" in the docs) would be better style:
```suggestion
            match = re.match(r'(\w+)([-+]*)$', area)
            if match is None:
                raise TraitError(f'"{area}" is not a valid string area.')
            region, modifier = match.groups()
```

Reply:
Gotcha, didn't think about other errors coming up.